winrm/client.go

```diff
@@ -32,6 +32,21 @@ func NewClient(endpoint *Endpoint, user, password string) (client *Client, err e
 // NewClient will create a new remote client on url, connecting with user and password
 // This function doesn't connect (connection happens only when CreateShell is called)
 func NewClientWithParameters(endpoint *Endpoint, user, password string, params *Parameters) (client *Client, err error) {
+
+	if endpoint.HTTPS == true {
+		if len(endpoint.Cert) == 0 || len(endpoint.Key) == 0 {
```

Comment:
I think you could use != "" or != nil depending on the type here.

It is more idiomatic than len in golang.